### ISSUE TITLE
apiKeyAuth returning "unauthorised"

### ISSUE THREAD

**Issue by poppet-uk** [intercom/intercom-node#329]
Please use the following template to submit your issue. Following this template will allow us to quickly investigate and help you with your issue. Please be aware that issues which do not conform to this template may be closed.

For feature requests please contact us at team@intercom.io

## Version info

-   intercom-node version: 3.1.5
-   Node version: 15.0.9

## Expected behavior
![Screenshot 2022-11-14 at 23 48 05](https://user-images.githubusercontent.com/116111932/201791705-301d37c3-a233-4b18-a8e5-01a42152d861.png)
Creating a new instance of the intercom Client using the "apiKeyAuth" should authenticate. "tokenAuth" works fine. 

## Actual behavior
Any request I make using the "apiKeyAuth" returns "unauthorized". 

## Steps to reproduce

1. create a new Client providing the apiKeyAuth params.
2. try to make a request to intercom

## Logs
```
Error: Unauthorized
    at new BadResponseError (/Users/drakemacbook2/Documents/Programming/cc/node_modules/lib/errors/badResponse.error.ts:10:9)
    at Client.checkOnErrorInResponse (/Users/drakemacbook2/Documents/Programming/cc/node_modules/lib/client.ts:258:16)
    at Client.<anonymous> (/Users/drakemacbook2/Documents/Programming/cc/node_modules/lib/client.ts:204:24)
    at step (/Users/drakemacbook2/Documents/Programming/cc/node_modules/intercom-client/dist/client.js:52:23)
    at Object.throw (/Users/drakemacbook2/Documents/Programming/cc/node_modules/intercom-client/dist/client.js:33:53)
    at rejected (/Users/drakemacbook2/Documents/Programming/cc/node_modules/intercom-client/dist/client.js:25:65)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by SeanHealy33** [comment]
Hi @poppet-uk 

I'm a little unsure of the issue here. The documented way of adding an api key is `const client = new Client({ tokenAuth: { token: 'my_token' } });` as documented in the readme. 

Could you let me know a little bit more about your use case

**Comment by SeanHealy33** [comment]
So I did some digging on this. 

`appAPIKey` in this context refers to a legacy token system we used before the Developer Hub was created (circa 2018). 

If you created your app in the dev hub and have a auth key from there then it uses TokenAuth exclusively. The api is returning  
`unauthorized` since your key does not exist on that workspace.

I'll keep this issue open though since we clearly need to document the different schemes of auth that the SDK supports and how they work.
